include/sys/sa_impl.h

```diff
@@ -114,6 +114,21 @@ typedef struct sa_idx_tab {
 	uint32_t	*sa_idx_tab;	/* array of offsets */
 } sa_idx_tab_t;
 
+#define	SA_IDX_ENTRIES		5
+
+typedef struct sa_idx_cache {
+	sa_idx_tab_t	*sac_idx[SA_IDX_ENTRIES];
+	sa_lot_t	*sac_lot[SA_IDX_ENTRIES];
+	int		sac_head;
```

Comment:
I think that `sac_lot` and `sac_idx` are accessed together, i.e. element `i` of both of these arrays together constitutes one entry in the cache.  Therefore we should use a struct to indicate this:

```
struct sa_idx_cache_entry {
    sa_lot_t *sice_lot;
    sa_idx_tab_t *sice_idx;
} sac_entries[SA_IDX_ENTRIES];
```